README.md

```diff
@@ -1,2 +1,160 @@
+
+To achieve this, this package disallows overly broad patterns like '**/*.js' and some of the more advanced features of
+the underlying library, such as regular expressions.  See below for full details.
+
+## `gpii.glob.findFiles(rootPath, includes, [excludes], [minimatchOptions])`
+
+* `rootPath`: A full or package-relative directory to scan for matching files.
+* `includes`: An `Array` of glob patterns that should be included in the results.
+* `excludes`: An optional `Array` of glob patterns that should be excluded from the results.
+* `minimatchOptions`: An optional `Object` detailing configuration options to be passed to [minimatch](https://github.com/isaacs/minimatch#options).
+* Returns: An `Array` of full paths to files that match the supplied glob patterns.
+
+## "glob" Patterns
+
+A "glob" pattern is a string that describes the path to one or more files.  It may contain single-asterisk wildcards
+such as `*.js`.  Single asterisks are only matched within a single path segment.  So, for example, `./test/*-node*.js`
+matches `./test/all-node-tests.js`, but not `./test/js/another-node-test.js`.
+
+"Glob" patterns may also use double-asterisks to indicate that any number of subdirectories may appear between one part
+of a pattern and the next.  So, for example, `./src/**/*.js` matches `./src/index.js` as well as
+`./src/js/other.js`.
+
+The underlying concept of a glob is powerful, but can lead to inefficient lookup strategies.  For the sake of
+performance, the following are not allowed in patterns used with this library:
+
+1. Patterns starting with `./**`  or `**`, which might require traversing all subdirectories before excludes can be
+   applied.
+2. Patterns that attempt to break out of the starting directory, i.e. that start with `../`.
+3. Patterns that use regular expressions to represent one or more parts of the path.
+4. Patterns that use the windows backslash separator in any part of the path.
+
+Patterns can be negated by prepending an exclamation point.  This mechanism allows you to define a more general rule and
+then identify one or more exceptions to that rule.   See below for examples.
+
+## Usage Examples
+
+Let's say you have a package called "my-package" whose structure looks roughly as diagrammed in this list:
+
+* (repository root)
+  * README.md
+  * index.js
+  * package.json
+  * .gitignore
+  * .eslintrc.json
+  * src
+    * lib
+      * forked-deps.js
+    * js
+      * index.js
+  * tests
+    * all-tests.js
+    * js
+      * test1.js
+      * test2.js
+
+Let's start by demonstrating includes.  Content can only be brought into scope by a regular (non-negated) include.
+
+```javascript
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+require("gpii-glob");
+
+// Let's assume that `fluid.module.resolvePath("%my-package")` resolves to `/source/my-package` for the purposes of
```

Comment:
Is there a use case for module-relative resolution? If the primary purpose of the library is as in a drop-in for grunt, isn't it most likely that people will just want to glob relative to content in their own package?

Reply:
I would like to leave that functionality in for use cases like bundling dependencies with our own source, such as I often do when preparing to serve up demos and other content from a static site.  That's just one example of where we might want to include content from other packages.

Also, it's not just a drop-in replacement for grunt.  We will use this ideally in gpii-testem as well to replace its one off globbing implementation.

Reply:
OK, sounds good to me